dace/sdfg/propagation.py

```diff
@@ -583,10 +920,66 @@ def propagate_memlets_state(sdfg, state):
     # 3. For each edge in the multigraph, collect results and group by array assigned to edge.
     #    Accumulate information about each array in the target node.
 
+    from dace.transformation.helpers import unsqueeze_memlet
+
     # First, propagate nested SDFGs in a bottom-up fashion
     for node in state.nodes():
         if isinstance(node, nodes.NestedSDFG):
-            propagate_memlets_sdfg(node.sdfg)
+            # Build a map of connectors to associated 'border' memlets inside
+            # the nested SDFG. This map will be populated with memlets once they
+            # get propagated in the SDFG.
+            border_memlets = {
+                'in': {},
+                'out': {},
+            }
+            for connector in node.in_connectors:
+                border_memlets['in'][connector] = None
+            for connector in node.out_connectors:
+                border_memlets['out'][connector] = None
+
+            # Propagate memlets inside the nested SDFG.
+            propagate_memlets_sdfg(node.sdfg, border_memlets)
+
+            # Make sure any potential NSDFG symbol mapping is correctly reversed
+            # when propagating out.
+            for direction in border_memlets:
+                for connector in border_memlets[direction]:
+                    border_memlet = border_memlets[direction][connector]
+                    if border_memlet is not None:
+                        border_memlet.substitute_symbol(node.symbol_mapping)
+
+            # Propagate the inside 'border' memlets outside the SDFG by
+            # offsetting, and unsqueezing if necessary.
+            for iedge in state.in_edges(node):
+                if iedge.dst_conn in border_memlets['in']:
+                    internal_memlet = border_memlets['in'][iedge.dst_conn]
+                    if internal_memlet is None:
+                        continue
+                    iedge._data = unsqueeze_memlet(
```

Comment:
`data` is a getter for `_data` decorated with `@property`. It cannot be directly written to, so it would either require a `@data.setter` annotated setter function or be written to `_data` directly.

Reply:
Addressed in ead2af4